### ISSUE TITLE
The sum of every dots' shap values should be zero or not?

### ISSUE THREAD

**Issue by sym823808458** [shap/shap#2477]
I'm working with the introductory documentation. I wanna get  overall importance of a feature. So I sum up the  shap values of every samples in the dataset and treated it as  overall importance of a feature.
However, someone told me the sum should be zero because of the defination of shap value.I don't know it's whether right or not?Or is it depend on the model algorithms?I am confused because I ran an example of MATLAB shapley algorithm, and the sum was so close to zero, while the example given here was not zero.I put my code here.

import xgboost
import shap

# train xgboost model on diabetes data:
X, y = shap.datasets.diabetes()
bst = xgboost.train({"learning_rate": 0.01}, xgboost.DMatrix(X, label=y), 100)

# explain the model's prediction using SHAP values on the first 1000 training data samples
shap_values = shap.TreeExplainer(bst).shap_values(X) 

print(np.average(shap_values, axis=0))

Any suggestions would be appreciated
